plugins/autosuspend.koplugin/main.lua

```diff
@@ -169,6 +171,57 @@ function AutoSuspend:onPreventStandby()
     self.standby_prevented = true
 end
 
+function AutoSuspend:setSuspendShutdownTimes(touchmenu_instance, title, setting, default_value)
+    local InfoMessage = require("ui/widget/infomessage")
+    local DateTimeWidget = require("ui/widget/datetimewidget")
+    -- About a minute, good for testing and battery life fanatics.
+    -- Just high enough to avoid an instant shutdown death scenario.
+    local min_time = 60
+    -- More than three weeks seems a bit excessive if you want to enable authoshutdown/suspend,
+    -- even if the battery can last up to three months.
+    local max_time_h = 21*24
+    local max_time = max_time_h * 3600
+    local duration_format = G_reader_settings:readSetting("duration_format", "classic")
+    local function show_info_message()
+        UIManager:show(InfoMessage:new{
+            text = T(_("%1: %2"), title,
+                util.secondsToClockDuration(duration_format, self[setting], true)),
+            timeout = 3,
+        })
+    end
+    UIManager:show(DateTimeWidget:new {
+        is_date = false,
+        hour = math.floor(self[setting] / 3600),
+        min = math.floor((self[setting] / 60) % 60),
+        hour_hold_step = 24,
+        min_hold_step = 10,
+        hour_max = max_time_h,
+        ok_text = _("Set timeout"),
+        title_text = title,
+        info_text = _("Set time in hours and minutes."),
+        callback = function(time)
+            self[setting] = math.floor((time.hour * 60 + time.min) * 60)
+            if self[setting] < min_time then
+                self[setting] = min_time
+            elseif self[setting] > max_time then
+                self[setting] = max_time
+            end
```

Comment:
If you like shortening as I do, `math.min(math.max(val, min), max)`.

Reply:
I don't necessarily like shortening like you do but I do think that one's clearer. ;-)

Reply:
Thanks, but I think the if-branches are more readable.